admob/src_java/com/google/firebase/admob/internal/cpp/BannerViewHelper.java

```diff
@@ -142,41 +142,57 @@ public void initialize(Activity activity) {
   /**
    * Destroy/deallocate the {@link PopupWindow} and {@link AdView}.
    */
-  public void destroy(final long callbackDataPtr) {
-    // If the Activity isn't initialized, there is nothing to destroy.
-    if (mActivity == null) {
-      completeBannerViewFutureCallback(callbackDataPtr,
-          ConstantsHelper.CALLBACK_ERROR_NONE,
-          ConstantsHelper.CALLBACK_ERROR_MESSAGE_NONE);
-      return;
-    }
-
-    // Stop any attempts to show the popup window.
-    synchronized (mPopUpLock) {
-      mPopUpRunnable = null;
-    }
+  public void destroy(final long callbackDataPtr, final boolean destructor_invocation) {
+    // If the Activity isn't initialized, or already Destroyed, then there's
+    // nothing to destroy.
+    if (mActivity != null) {
+      mActivity.runOnUiThread(
+          new Runnable() {
+            @Override
+            public void run() {
+              // Stop any attempts to show the popup window.
+              synchronized (mPopUpLock) {
+                mPopUpRunnable = null;
+              }
 
-    if (mAdView != null) {
-      mAdView.setAdListener(null);
-      mAdView.setOnPaidEventListener(null);
-      mAdView.destroy();
-      mAdView = null;
-    }
+              if (mAdView != null) {
+                mAdView.setAdListener(null);
+                mAdView.setOnPaidEventListener(null);
+                mAdView.destroy();
+                mAdView = null;
+              }
 
-    synchronized (mPopUpLock) {
-      if (mPopUp != null) {
-        mPopUp.dismiss();
-        mPopUp = null;
+              synchronized (mPopUpLock) {
+                if (mPopUp != null) {
+                  mPopUp.dismiss();
+                  mPopUp = null;
+                }
+              }
+              synchronized (mBannerViewLock) {
+                if (destructor_invocation == false) {
+                  notifyBoundingBoxChanged(mBannerViewInternalPtr);
```

Comment:
Don't invoke the bounding box listener if this destroy() invocation originated from the C++ object destructor.  The bounding box listener won't be there anymore.